holopy/core/tests/test_utils_math.py

```diff
@@ -238,41 +239,167 @@ def test_rotation_matrix_degrees():
+        self.assertTrue(ensure_scalar(np.array([1])) == 1)
 
 @attr("fast")
 def test_mkdir_p():
     tempdir = tempfile.mkdtemp()
     mkdir_p(os.path.join(tempdir, 'a', 'b'))
     mkdir_p(os.path.join(tempdir, 'a', 'b'))
     shutil.rmtree(tempdir)
+
+
+class TestDictionaryUtils(unittest.TestCase):
+    input_dict = {'a':1, 'b':2, 'c':3, 'd':4}
+    update_dict = {'c':5, 'd':None, 'e':6}
+
+    @attr("fast")
+    def test_dict_without(self):
+        output_dict = dict_without(self.input_dict, ['a','d','e'])
+        self.assertTrue(self.input_dict == {'a':1, 'b':2, 'c':3, 'd':4})
+        self.assertTrue(output_dict == {'b':2, 'c':3})
+
+    @attr("fast")
+    def test_updated_basic(self):
+        output_dict = updated(self.input_dict, self.update_dict)
+        self.assertTrue(self.input_dict == {'a':1, 'b':2, 'c':3, 'd':4})
+        self.assertTrue(output_dict == {'a':1, 'b':2, 'c':5, 'd':4, 'e':6})
+
+    @attr("fast")
+    def test_updated_keep_None(self):
+        output_dict = updated(self.input_dict, self.update_dict, False)
+        self.assertTrue(self.input_dict == {'a':1, 'b':2, 'c':3, 'd':4})
+        self.assertTrue(output_dict == {'a':1, 'b':2, 'c':5, 'd':None, 'e':6})
+
+    @attr("fast")
+    def test_updated_from_kw(self):
+        output_dict = updated(self.input_dict, b=7, c=None, e=8)
+        self.assertTrue(self.input_dict == {'a':1, 'b':2, 'c':3, 'd':4})
+        self.assertTrue(output_dict == {'a':1, 'b':7, 'c':3, 'd':4, 'e':8})
+
+    @attr("fast")
+    def test_kw_takes_priority(self):
+        output_dict = updated(self.input_dict, self.update_dict, b=7, e=8)
+        self.assertTrue(self.input_dict == {'a':1, 'b':2, 'c':3, 'd':4})
+        self.assertTrue(output_dict == {'a':1, 'b':7, 'c':5, 'd':4, 'e':8})
+
+
+class TestRepeatSingDims(unittest.TestCase):
+    input_dict = {'x':[0], 'y':[1], 'z':[0,1,2]}
+    # these tests compare dictionaries containing numpy arrays
+    # using np.testing.assert_equal to avoid errors.
+
+    @attr("fast")
+    def test_all_keys(self):
+        output_dict = {'x':np.array([0, 0, 0]), 'y':np.array([1, 1, 1]),
+                      'z':[0, 1, 2]}
+        assert_equal(repeat_sing_dims(self.input_dict), output_dict)
+
+    @attr("fast")
+    def test_input_isnt_modified(self):
+        repeat_sing_dims(self.input_dict)
+        self.assertTrue(self.input_dict == {'x':[0], 'y':[1], 'z':[0,1,2]})
+
+    @attr("fast")
+    def test_repeat_some_keys(self):
+        output_dict ={'x':np.array([0,0,0]), 'y':[1], 'z':[0, 1, 2]}
+        repeated = repeat_sing_dims(self.input_dict, ['x', 'z'])
+        assert_equal(repeated, output_dict)
+
+    @attr("fast")
+    def test_nothing_to_repeat(self):
+        repeated = repeat_sing_dims(self.input_dict, ['x', 'y'])
+        self.assertTrue(repeated == self.input_dict)
+
+class TestChoosePool(unittest.TestCase):
+    @attr("fast")
+    def test_custom_pool(self):
+        class DummyPool():
```

Comment:
It might be nice to move the ```DummyPool``` outside this test case, so that multiple tests can use it easily if desired.